erb2epp

```diff
@@ -1,77 +1,106 @@
-#!/usr/bin/ruby
+#!/usr/bin/env ruby
 
 require 'temple'
 require 'ripper'
 
 # Rewrite code from Ruby to EPP
-#
-# This comes down to replacing @var with $var
-def rewrite_code(code)
-  tokens = Ripper.lex(code)
+class Erb2epp
+  def initialize
+    @vars_found = []
+  end
 
-  # First pass: rewrite
-  tokens.each do |token|
-    # [[lineno, column], type, token, state]
-    _, type, value, _state = token
-    case type
-    when :on_ivar
-      token[2] = value.gsub(/@([a-z][A-Za-z0-9_]*)/, '$\1')
-      # TODO: Store variable name somewhere to build a list of inputs
-    when :on_op
-      case value
-      when '||'
-        token[2] = 'or'
-      when '&&'
-        token[2] = 'and'
-      end
-    when :on_kw
-      case value
-      when 'end'
-        token[2] = '}'
-      when 'if'
-        # TODO: append { somewhere
+  # Rewrite a line of code
+  def rewrite_code(code)
+    tokens = Ripper.lex(code)
+    res = []
+    ops = {}
+
+    # First pass: rewrite
+    tokens.each do |token|
+      # [[lineno, column], type, token, state]
+      _, type, value, _state = token
+      r = value
+
+      case type
+      when :on_ivar
+        r = value.gsub(%r{@([a-z][A-Za-z0-9_]*)}, '$\1')
+        @vars_found.push r
+      when :on_op
+        case value
+        when '||'
+          r = 'or'
+        when '&&'
+          r = 'and'
+        end
+      when :on_kw
+        case value
+        when 'if'
+          ops['if'] = true
```

Comment:
I wonder if you should look ahead in the tokens to find the right place. For example:
```ruby
if x
  y = 1
end
```

This results in
```
[
 [[1, 0], :on_kw, "if", BEG],
 [[1, 2], :on_sp, " ", BEG],
 [[1, 3], :on_ident, "x", CMDARG],
 [[1, 4], :on_nl, "\n", BEG],
 [[2, 0], :on_sp, "  ", BEG],
 [[2, 2], :on_ident, "y", CMDARG],
 [[2, 3], :on_sp, " ", CMDARG],
 [[2, 4], :on_op, "=", BEG],
 [[2, 5], :on_sp, " ", BEG],
 [[2, 6], :on_int, "1", END],
 [[2, 7], :on_nl, "\n", BEG],
 [[3, 0], :on_kw, "end", END],
]
```

So you can scan for `on_nl` and know that it should be inserted there.

For that you probably need to separate the rewriting into its own function.